dashboard/app/models/levels/blockly.rb

```diff
@@ -564,16 +564,26 @@ def shared_functions
     end.join
   end
 
-  # Display translated custom block text
+  # Display translated custom block text and options
   def localized_shared_blocks(level_objects)
     return nil if level_objects.blank?
+
     level_objects_copy = level_objects.deep_dup
     level_objects_copy.each do |level_object|
       next if level_object.blank?
       block_text = level_object[:config]["blockText"]
-      if block_text.present?
-        level_object[:config]["blockText"] = I18n.t("data.blocks.#{level_object[:name]}.text")
+      next if block_text.blank?
+      level_object[:config]["blockText"] = I18n.t("data.blocks.#{level_object[:name]}.text")
+      options_keys = block_text.scan(/\{(.*?)\}/).flatten
```

Comment:
this seems like an unnecessary optimization

Reply:
@Hamms, I agree but I got a lint error to use next if instead of the if block.

Reply:
no, `next` is correct in this case, I'm referring to scanning the block text to get the options keys. We shouldn't care whether or not the keys are used in the block text when deciding whether or not to translate them